astropy/config/tests/test_configs.py

```diff
@@ -16,6 +16,19 @@
 from astropy.utils.exceptions import AstropyDeprecationWarning
 
 
+OLD_CONFIG = {}
+
+
+def setup_module():
+    OLD_CONFIG.clear()
+    OLD_CONFIG.update(configuration._cfgobjs)
```

Comment:
The need to call "hidden" stuff might indicate that we should expose the API properly? But that is discussion for another day.

Reply:
Yes, or one should have a context manager option that ensures the configuration gets returned in its original state. But I'd indeed like to keep that for another PR.